app/src/main/kotlin/moe/koiverse/archivetune/widget/PlayerWidgetProvider.kt

```
@@ -0,0 +1,308 @@
+        appWidgetIds: IntArray
+    ) {
+        val sessionToken = SessionToken(context, ComponentName(context, MusicService::class.java))
+        val controllerFuture = MediaController.Builder(context, sessionToken).buildAsync()
+
+        controllerFuture.addListener({
+            try {
+                val controller = controllerFuture.get()
+                updateWidgetsFromController(context, appWidgetManager, appWidgetIds, controller)
+                // Release controller after use to prevent leaks
+                MediaController.releaseFuture(controllerFuture)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)